### PR TITLE
Fix courses sort order

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
This is the bug where courses are not ordered correctly in publish. Here is the card:
https://trello.com/c/XY3YkS5d/800-courses-are-no-longer-ordered-alphabetically-in-publish

### Changes proposed in this pull request
Added sort order to the dbcontext
Added test that would have flushed out this bug
### Guidance to review
The original code had a sort order but this had been over-ridden by the addition of includes that caused ef to ignore the ordering.
The order by has now been moved to the linq in the context.

image

image

image

### CODE REVIEW

**Comment by fofr** [comment]
When courses are grouped by the accredited provider, are they alphabetical within that group?

File: src/ManageCourses.Domain/DatabaseAccess/ManageCoursesDbContext.cs [hunk 1]
```diff
@@ -207,11 +207,13 @@ public List<Course> GetCoursesByProviderCode(string providerCode, string email)
...
                     $"join \"user\" u on ou.user_id = u.id " +
                     $"where lower(i.provider_code)=lower(@providerCode) " +
-                    $"and lower(u.email)=lower(@email) order by c.name", new NpgsqlParameter("providerCode", providerCode), new NpgsqlParameter("email", email))
+                    $"and lower(u.email)=lower(@email)", new NpgsqlParameter("providerCode", providerCode), new NpgsqlParameter("email", email))
                 .Include(x => x.Provider)
                 .Include(x => x.CourseSubjects).ThenInclude(x => x.Subject)
                 .Include(x => x.AccreditingProvider)
                 .Include(x => x.CourseSites).ThenInclude(x => x.Site)
+                .OrderBy(x => x.Name)//order by doesn't work as part of the sql string
+                                     //it's an ef issue caused by the includes
```
Comment (timabell) [inline src/ManageCourses.Domain/DatabaseAccess/ManageCoursesDbContext.cs]: I don't think I'd bother adding this comment personally

**Comment by MartinBentley** [comment]
> When courses are grouped by the accredited provider, are they alphabetical within that group?

Yes they are. I've attached a screenshot for this.

**Comment by timabell** [comment]
[Courses are no longer ordered alphabetically in Publish](https://trello.com/c/XY3YkS5d/800-courses-are-no-longer-ordered-alphabetically-in-publish)

**Pull request merged**